lib/TH/generic/THTensor.c

```diff
@@ -285,6 +285,57 @@ void THTensor_(resize5d)(THTensor *self, long size0, long size1, long size2, lon
   THTensor_(resizeNd)(self, 5, size, NULL);
 }
 
+void THTensor_(expandNd)(THTensor *tensor, long *sizes, ptrdiff_t ndim) {
+  THArgCheck(tensor->nDimension > 0, 0, "can't expand an empty tensor");
+  long numUnsqueezed = ndim - tensor->nDimension;
+
+  long *expandedSizes = THAlloc(sizeof(long)*ndim);
+  long *expandedStrides = THAlloc(sizeof(long)*ndim);
+
+  for (long i = numUnsqueezed; i < ndim; ++i) {
```

Comment:
This doesn't look correct to me.  Consider the case where dims(tensor) == 1 and ndim = 2.  Then the first time through the loop i = 1, and we try to get THTensor_(size)(tensor, 1), but there is only a dim 0.